### PR TITLE
Update Jetpack Cloud Download Styling

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request

* Add new Jetpack Cloud `RewindFlow` component to be used for both restore and download flow
* Implement the download half of `RewindFlow`
* Remove all existing download-flow specific components
* Add and refine download related selectors to clarify flow

The Download Flow

#### Testing instructions

1. Navigate to `/backups/` for a site that has Jetpack backups
2. Pick a backup to download, and click "Download Backup"
3. Verify you are on a page that matched "1. Confirm & Configure Download" above
4. Make at least one change to the download configuration, noting what it is, and open network tools.
5. Click "Create downloadable file"
6. Find the POST to `/wpcom/v2/sites/:sitId/rewind/downloads` in the network tools and verify that the body matched the configuration demo step 4.
7. Verify the in-progress screen matches "2. Download In-Progress" above
8. Verify that eventually the screen goes to "3. Download Ready"
9. Verify that the download is ready
10. Verify you received an email for the site saying the download is ready
11. Reload the page and verify after briefly seeing "1. Confirm & Configure Download" the page jumps to "3. Download Ready", since the download is already ready.

#### Next Steps

1. Better error handling. Anything in the flow that is not expected will result in this screen:

2. Bring the restore flow in using the `RewindFlow` component to reduce repeated style code and components.

3. Rework of the underlying request for downloads. It returns 0-10 of the most recent backups, but we only grab the first if it exists. Handling for zero returned downloads is poor. We would want to
    a. Get all returned backups to reduce re-doing download creation
    b. improve the handling of 0 returned downloads. This will also allow request progress to avoid the "jump" from step 11

### CODE REVIEW

File: client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx
Comment (a8ci18n) [inline client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **2** times:
`translate( '{{strong}}%(downloadTimestamp)s{{/strong}} is the selected point to create a download backup of.' )` *ES Score: 8*

**Comment by noahtallen** [comment]
guessing the review pings are from a bad rebase :)

**Comment by cbauerman** [comment]
> guessing the review pings are from a bad rebase :)

🤦‍♂  didn't retarget the PR before the rebase, that is embarrassing

File: client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx
Comment (enejb) [inline client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx]: Should this also be translated?

File: client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx
Comment (enejb) [inline client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx]: Should we tell the user the email address that will be used?

File: client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx
Comment (enejb) [inline client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx]: This doesn't seem relevant. 
Instead of "Check your email" heading we should probably something different. 

Like "Email Sent" for example.
...

File: client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx
Comment (enejb) [inline client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx]: This div seems to be repeated quite a few times. 
Can we make it a component?

File: client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx
Comment (enejb) [inline client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx]: Since this ProgressBar is only used once do we need to make it a component?

**Comment by enejb** [comment]
Thanks for working on this. I spend sometime going over the flow and have some general feedback that I noticed going though the flow. 

I was able to download the zip files as expected. 

I think it is a great improvement over what what we currently have and we should address some of the issues that I found in a different PRs. 

Few things I noticed. I was testing this in latest FF on a mac. 

**Create Download Page**
Screen Shot 2020-03-24 at 11 40 22 AM

- The checkboxes don't seem to be aligned with the labels. 
- The More Info is a external link. (It is also bold the Figma file shows it no be regular)
We should probably label it as such. It was a bit unexpected for me to take me off site. Maybe we can provide some more information inline regarding the different things that the user is choosing. Maybe even server paths. (This is my dev brain talking for things like the plugins folder, themes, uploads) 
- There is a comma missing after the Date in the first line. 


**Creating Downloadable**
Screen Shot 2020-03-24 at 11 41 50 AM

 This looked great to me.
- It would be great to be more specific. Like if you only selected plugins folder that we would tell the user that we creating a downloadable backup of the plugins. Also we could drop the "Currently " at the beginning of. Maybe the title should be something like "Packaging up exmple.com from (date)..." 
Currently there is quite a bit of redundancy with the bottom description with the title.

IMHO We should also link to support from this page. 

Backup Download page.
Screen Shot 2020-03-24 at 11 41 11 AM

I think this is great. 

- I think we should highlight that we send the email. Vs Tell the user to check the email. 
- Looking at the email It seems that the download link expires and is only valid for an hour. Should we update this UI to reflect that? 
- It would be great to provide a support link here as well. 
- the file size is currently missing from the UI. 
- It would be great to tell the user what the download contains. (What did they select? Not detailed info what is in the backup)


Over all I would love to have this PR broken up into smaller PRs. That would make it much easier to review.

**Review by rcoll — CHANGES_REQUESTED** [review]
This looks really really good as is. Two things:
- I agree with @enejb's comments, address what you can quickly, and let's make Asana cards for the rest
- This branch needs a rebase to check styles. Looks like the branch is missing the merges that brought in the proper "base styling". Probably best to make sure there is no problem there after the rebase.

File: client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx
Comment (cbauerman) [reply client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx]: tagging design for review

File: client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx
Comment (cbauerman) [reply client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx]: tagging design for review

File: client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx
Comment (cbauerman) [reply client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx]: > Add new Jetpack Cloud RewindFlow component to be used for both restore and download flow
I will circle back around and reuse it for the nearly-identical restore flow

File: client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx
Comment (cbauerman) [reply client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx]: I was waiting until I had both flows to make a final decision on if these four line elements needed their own components

**Comment by cbauerman** [comment]
@enejb 

1. In this PR tweaked the checkboxes and fixed some other style issues

2. For the componentization/re-use of components I will follow up in PR that uses the `RewindFlow` components for Restore

3. For parts of the design that are not addressed ( mainly the file size ) I will also follow up with other PR as a  "progressive enhancement"

4. For everything else related to the text/design we should follow up with our designers. I will summarize the points you brought up

**Pull request merged**
